### PR TITLE
[Event Hubs Client] Track Two: Fifth Preview (Documentation Updates)

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

Updates for the README and CHANGELOG files to align with the new client hierarchy.  Additionally, some of the sample descriptions have been tweaked.

# Last Upstream Rebase

Thursday, October 31, 9:51am (EDT)

# Related and Follow-Up Issues

- [Release Event Hubs Track 2 Library Preview 5 for .Net ](https://github.com/Azure/azure-sdk-for-net/issues/8190) (#8190) 
- [Review and update documentation ](https://github.com/Azure/azure-sdk-for-net/issues/8191) (#8191)

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,51 @@
...
+
+#### Organization and naming
+
+- The early stages of a large refactoring of the Event Hub client type hierarchy is complete;  Event Hub producers and consumers are now stand-alone clients that can be created independently, and which may or may not share a connection to the Event Hubs service at the discretion of the developer.    
+
+  **_Please Note:_** These changes are part of a larger effort that is currently in process.  The client types have not yet been refactored to remove affinity to a specific partition which leaves some operations feeling somewhat awkward with the new structure.
+  
+#### Bug fixes and foundation improvements
+
+- Client types using Azure identity for authorization should now be work properly; a bug was fixed to allow the proper authorization scopes to be requested.
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md]: `be work` => `be working`?

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 2]
```diff
@@ -35,30 +35,20 @@ To quickly create the needed Event Hubs resources in Azure and to receive a conn
...
-```
-
 ## Key concepts
 
-- An **Event Hub client** is the primary interface for developers interacting with the Event Hubs client library, allowing for inspection of Event Hub metadata and providing a guided experience towards specific Event Hub operations such as the creation of producers and consumers.
+- An **Event Hub client** is the primary interface for developers interacting with the Event Hubs client library.  There are several different Event Hub clients, each dedicated to a specific use of Event Hubs, such as publishing or consuming events.
 
-- An **Event Hub producer** is a source of telemetry data, diagnostics information, usage logs, or other log data, as part of an embedded device solution, a mobile device application, a game title running on a console or other device, some client or server based business solution, or a web site.  
+- An **Event Hub producer** is type of client that serves as a source of telemetry data, diagnostics information, usage logs, or other log data, as part of an embedded device solution, a mobile device application, a game title running on a console or other device, some client or server based business solution, or a web site.
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/README.md]: `is type` => `is a type`

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 3]
```diff
@@ -105,82 +94,86 @@ await using (EventHubProducer producer = client.CreateProducer())
...
-    string consumerGroup = EventHubConsumer.DefaultConsumerGroupName;
-    EventPosition startingPosition = EventPosition.Earliest;
+    partitionId = (await client.GetPartitionIdsAsync()).First();
+}
     
-    await using (EventHubConsumer consumer = client.CreateConsumer(consumerGroup, firstPartition, startingPosition))
-    {
-        using CancellationTokenSource cancellationSource = new CancellationTokenSource();
-        cancellationSource.CancelAfter(TimeSpan.FromSeconds(45));
+await using (var consumer = new EventHubConsumerClient(consumerGroup, partitionId, startingPosition, connnectionString, eventHubName))
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/README.md]: typo: `connnection`

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 4]
```diff
@@ -105,82 +94,86 @@ await using (EventHubProducer producer = client.CreateProducer())
...
     //
-    // It is important to note that the processor does not own the EventHubClient that was passed into it.  
-    // You are responsible for ensuring that it is disposed after processing.  It is also important that it
-    // not be closed or disposed during the time that the EventProcessor is running.  
-    //
-    // In this example, we'll stop processing after five minutes.
-    
+    // In this example, we'll stop processing after two minutes has elapsed.
+        
     await Task.Delay(TimeSpan.FromMinutes(5));
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/README.md]: `two minutes` in the comment and `FromMinutes(5)` in the code.

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 5]
```diff
@@ -200,7 +193,6 @@ This indicates that the Event Hubs service did not respond to an operation withi
 #### Message Size Exceeded
 
 Event data, both individual and in batches, have a maximum size allowed.  This includes the data of the event, as well as any associated metadata and system overhead.  The best approach for resolving this error is to reduce the number of events being sent in a batch or the size of data included in the message.  Because size limits are subject to change, please refer to [Azure Event Hubs quotas and limits](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-quotas) for specifics.  
-
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/README.md]: Empty line should be kept.

**Review by kinelski — APPROVED** [review]
LGTM. Left some minor comments.

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample12_ConsumeEventsWithEventProcessor.cs [hunk 6]
```diff
@@ -127,10 +127,14 @@ await using (var eventProcessor = new EventProcessorClient(EventHubConsumerClien
...
-                    // It is considered responsibility of the developer writing the handler code to ensure that proper exception handling practices are
-                    // followed.
+                    // Any exception which occurs as a result of the event processor itself will be passed to
+                    // this delegate so it may be handled.  The processor will continue to process events if
+                    // it is able to unless this handler explicitly requests that it stop doing so.
+                    //
+                    // It is important to note that this does not include exceptions during event processing; those
+                    // are considered responsibility of the developer implementing the event processing handler.  It
+                    // is, therefore, highly encouraged that best practices for exception handling practices are
+                    // followed with that delegate.
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample12_ConsumeEventsWithEventProcessor.cs]: Yeah, I'm writing a few of these for "what the design calls for."  We'll be inconsistent for a bit.

**Pull request merged**